.github/workflows/release.yaml

```diff
@@ -11,13 +11,71 @@ permissions:
   contents: read
 env:
   ARTIFACT_NAME: artifact-ubuntu-latest-python-3.10
+  USER_NAME: github-actions # The default user name for release.
+  USER_EMAIL: github-actions@github.com # The default email address for release.
 
 jobs:
+  check:
+    uses: ./.github/workflows/build.yaml
+    permissions:
+      contents: read
+    with:
+      is_release_build: false
+  bump:
+    needs: check
+    if: ${{ github.ref == 'refs/heads/main' && !startsWith(github.event.commits[0].message, 'bump:') }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+    - name: Harden Runner
+      uses: step-security/harden-runner@dd2c410b088af7c0dc8046f3ac9a8f4148492a95 # v1.4.5
+      with:
+        egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+    - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+      with:
+        fetch-depth: 0
+        token: ${{ secrets.PERSONAL_TOKEN }}
+    - name: Set up Python
+      uses: actions/setup-python@b55428b1882923874294fa556849718a1d7f2ca5 # v4.2.0
+      with:
+        python-version: '3.10'
+    - name: Set up Commitizen
+      run: |
+        pip install --upgrade pip wheel
+        pip install 'commitizen ==2.32.1'
+    - name: Create changelog and bump
+      run: |
+        # Set up the GitHub user and email as author for the release commit.
+        git config --global user.name $USER_NAME
+        git config --global user.email $USER_EMAIL
+        git config --list --global
+        # First generate the CHANGELOG diff to use as release notes.
+        mkdir -p dist
```

Comment:
Technically we don’t need `dist/` here anymore, do we?

Reply:
We actually need it because we don't call `make setup` here for efficiency concerns.